src/EditorFeatures/CSharpTest/CodeActions/IntroduceVariable/IntroduceVariableTests.cs

```diff
@@ -7813,5 +7813,74 @@ void M()
     }
 }");
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsIntroduceVariable)]
+        [WorkItem(47772, "https://github.com/dotnet/roslyn/issues/47772")]
+        public async Task DonotIntroduceConstantForConstant_Local()
+        {
+            await TestMissingAsync(
+@"
+class C
+{
+    void M()
+    {
+        const int foo = [|10|];
+    }
+}
+");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsIntroduceVariable)]
+        [WorkItem(47772, "https://github.com/dotnet/roslyn/issues/47772")]
+        public async Task DonotIntroduceConstantForConstant_Member()
+        {
+            await TestMissingAsync(
+@"
+class C
+{
+    const int foo = [|10|];
+}
+");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsIntroduceVariable)]
+        [WorkItem(47772, "https://github.com/dotnet/roslyn/issues/47772")]
+        public async Task DonotIntroduceConstantForConstant_Parentheses()
+        {
+            await TestMissingAsync(
+@"
+class C
+{
+    const int foo = ([|10|]);
+}
+");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsIntroduceVariable)]
+        [WorkItem(47772, "https://github.com/dotnet/roslyn/issues/47772")]
+        public async Task DonotIntroduceConstantForConstant_NotForSubExpression()
```

Comment:
this name is just confusing.  having double negatives is not a good thing in names :)